bluesky_queueserver/manager/qserver_cli.py

```diff
@@ -15,162 +15,690 @@
+        the item (value is a string representing UID). Empty dictionary if no address
+        is found.
+    list(str)
+        List of the remaining parameters
+
+    Raises
+    ------
+    IndexError
+        Insufficient number of parameters are provided (less than 1)
+    """
+    n_used = 0
+
+    pos, uid = None, None
+    if params[0] in ("front", "back"):
+        pos = params[0]
+        n_used = 1
+    else:
+        try:
+            pos = int(params[0])
+            n_used = 1
+        except Exception:
+            ...
+
+        if pos is None:
+            uid = params[0]
+            n_used = 1
+
+    if pos is not None:
+        addr_param = {"pos": pos}
+    elif uid is not None:
+        addr_param = {"uid": uid}
+    else:
+        addr_param = {}
+    return addr_param, params[n_used:]
+
+
+def extract_destination_address(params):
+    """
+    Extract 'destination' item address (index or UID) from the list of parameters. Returns
+    the list of remaining parameters. The source address is represented by 1 or 2 parameters:
+    if 1st parameter is a keywords ``front`` ``back`` or integer number, then the parameter
+    represents a positional address (index); if the 1st parameter is a keyword ``before`` or
+    ``after``, the 2nd parameter is considered to represent item UID (string). If the 1st
+    parameter can not be converted to ``int`` or equal to one of the keywords, it is considered
+    that the address is not found
+
+    Parameters
+    ----------
+    params : list(str)
+         List of parameters. The 1st and optionally 2nd parameter are interpreted as a destination
+         address.
+
+    Returns
+    -------
+    dict
+        Dictionary that contains destination address. Elements: ``pos`` - positional address
+        (value is int or a string from the set ``front``, ``back``), ``before_uid`` or ``after_uid``
+        - uid of the item preceding or following the destination for the item in the queue (value
+        is a string representing UID).  Empty dictionary if no address is found.
+    list(str)
+        List of the remaining parameters.
+
+    Raises
+    ------
+    IndexError
+        Insufficient number of parameters are provided.
+    """
+    n_used = 0
+
+    pos, uid, uid_key = None, None, None
+    if params[0] in ("front", "back"):
+        pos = params[0]
+        n_used = 1
+    elif params[0] in ("before", "after"):
+        uid = params[1]
+        # Keys are "before_uid" and "after_uid"
+        uid_key = f"{params[0]}_uid"
+        n_used = 2
+    else:
+        try:
+            pos = int(params[0])
+            n_used = 1
+        except Exception:
+            ...
+
+    if pos is not None:
+        addr_param = {"pos": pos}
+    elif uid is not None:
+        addr_param = {uid_key: uid}
+    else:
+        addr_param = {}
+    return addr_param, params[n_used:]
+
+
+def format_list_as_command(params):
```

Comment:
I was thinking about it. Here it is used to convert parameters from command string represented as a list back to a string so that it could be printed in error message. So may be it makes sense to leave it.

Reply:
OK, let's keep it. The name is not misleading at least.